src/Core/Content/Product/Cart/ProductCartProcessor.php

```diff
@@ -234,11 +234,7 @@ private function validateStock(LineItem $item, Cart $cart, LineItemCollection $s
             $steps = $item->getQuantityInformation()->getPurchaseSteps() ?? 1;
         }
 
-        if ($item->getQuantity() < $minPurchase) {
-            $item->setQuantity($minPurchase);
-        }
-
-        if ($available <= 0 || $available < $minPurchase) {
```

Comment:
$minPurchase is at least 1, so the left side of the condition is always true and can be removed